client/src/components/SearchBar.js

```diff
@@ -2,14 +2,15 @@ import React from "react";
 
 const SearchBar = ({ videoData, setVideoData }) => {
   const handleSearch = (e) => {
-    const searchVideoData = videoData.filter((video) =>
-      video.title.toLowerCase().includes(e.target.value.toLowerCase())
-    );
-    setVideoData(searchVideoData);
+    if (e.target.value.length > 0) {
+      const searchVideoData = videoData.filter((video) =>
+        video.title.toLowerCase().includes(e.target.value.toLowerCase())
+      );
+      setVideoData(searchVideoData);
+    }
```

Comment:
this is close to resetting the videos when the input string is empty - you might want to look at adding in an else condition, so that if a user backspaces to return to an empty string, it shows all videos (for example if you imported your original video data in this component, you could setVideoData back to this original data if e.target.value.length became 0.